nltk/text.py

```diff
@@ -530,16 +534,50 @@ def dispersion_plot(self, words):
         from nltk.draw import dispersion_plot
 
         dispersion_plot(self, words)
-
-    def generate(self, words):
-        """
-        Issues a reminder to users following the book online
-        """
-        import warnings
-
-        warnings.warn(
-            'The generate() method is no longer available.', DeprecationWarning
-        )
+        
+    def build_ngram_lm(self, tokenized_sents, n=3):
+        train_data, padded_sents = padded_everygram_pipeline(n, tokenized_sents)
+        model = MLE(order=n)
+        model.fit(train_data, padded_sents)
+        return model
+        
+    def generate(self, length=100, text_seed=None, random_seed=42):
+        """
+        Print random text, generated using a trigram language model.
+        See also `help(nltk.lm)`.
+        
+        :param length: The length of text to generate (default=100)
+        :type length: int
+        
+        :param text_seed: Generation can be conditioned on preceding context.
+        :type text_seed: list(str)
+        
+        :param random_seed: A random seed or an instance of `random.Random`. If provided,
+        makes the random sampling part of generation reproducible.
+        :type random_seed: int
+        
+        """
+        # Create the model when using it the first time.
+        self.tokenized_sents = [sent.split(' ') for sent in 
+                                sent_tokenize(' '.join(self.tokens))]
+        if not hasattr(self, 'trigram_model'):
```

Comment:
Similar concern here: do people expect that an ngram model gets trained in the background if they want to generate from a text? This is where I'm not familiar enough with how this class gets used to judge what folks will find surprising or not.

Reply:
It's because most people using the `generate()` function is following the tutorial on https://www.nltk.org/book/ch01.html so I think it's okay since the `Text` object are not using in other places.

Reply:
Alright, if it's really mostly for demo purposes (though why would you demo something that isn't used anywhere else?) then I guess we can leave this stuff...

Reply:
Change it's to a private attribute convention `_ trigram_model`.